File: C/2180-count-integers-with-even-digit-sum/main.go
package main

func countEven(num int) int {
	buf := make([]bool, num+1)
	buf[0] = true
	for i := 1; i <= num; i++ {
		buf[i] = buf[i/10] == ((i%10)%2 == 0)
	}
	cnt := 0
	for i := 1; i <= num; i++ {
		if buf[i] {
			cnt++
		}
	}
	return cnt
}

File: C/2180-count-integers-with-even-digit-sum/main.rs
struct Solution {}

impl Solution {
  pub fn count_even(num: i32) -> i32 {
    (2..=num)
      .map(|n| {
        let mut sum: i32 = 0;
        let mut m = n;
        while m > 0 {
          sum += m % 10;
          m /= 10;
        }
        (sum + 1) % 2
      })
      .sum()
  }
}
